packages/crdt/src/crdt/merkle.ts

```diff
@@ -52,10 +58,11 @@ function insertKey(trie: TrieNode, key: string, hash: number): TrieNode {
     return trie;
   }
   const c = key[0];
-  const n = trie[c] || {};
+  const t = isNumberTrieNodeKey(c) ? trie[c] : undefined;
+  const n = t || {};
```

Comment:
Functionally this again is exactly the same as before. I'm just making types more strict and allowing them to pass.